Sources/Rownd/Models/Context/Passkeys.swift

```diff
@@ -0,0 +1,137 @@
+//
+//  Auth.swift
+//  framework
+//
+//  Created by Matt Hamann on 7/8/22.
+//
+
+import Foundation
+import UIKit
+import ReSwift
+import ReSwiftThunk
+import AnyCodable
+import Get
+
+public struct PasskeyRegistration: Hashable {
+    public var id: String?
+}
+
+extension PasskeyRegistration: Codable {
+    enum CodingKeys: String, CodingKey {
+        case id
+    }
+}
+
+public struct PasskeyState: Hashable {
+    public var isLoading: Bool = false
+    public var isInitialized: Bool = false
+    public var isErrored: Bool = false
+    public var errorMessage: String?
+    public var registration: [PasskeyRegistration]? = []
+}
+
+extension PasskeyState: Codable {
+    public enum CodingKeys: String, CodingKey {
+        case registration
+    }
+
+    public func get() -> PasskeyState {
+        return self
+    }
+}
+
+struct SetPasskeyState: Action {
+    public var payload = PasskeyState()
+}
+
+struct SetPasskeyLoading: Action {
+    var isLoading: Bool
+}
+
+struct SetPasskeyInitialized: Action {
+    var isInitialized: Bool
+}
+
+struct SetPasskeyRegistration: Action {
+    var payload: [PasskeyRegistration]
+}
+
+struct SetPasskeyError: Action {
+    var isErrored: Bool = true
+    var errorMessage: String
+}
+
+func passkeyReducer(action: Action, state: PasskeyState?) -> PasskeyState {
+    var state = state ?? PasskeyState()
+    
+    switch action {
+    case let action as SetPasskeyState:
+        state = action.payload
+    case let action as SetPasskeyLoading:
+        state.isLoading = action.isLoading
+    case let action as SetPasskeyInitialized:
+        state.isInitialized = action.isInitialized
+    case let action as SetPasskeyRegistration:
+        state.isInitialized = true
+        state.registration = action.payload
+    default:
+        break
+    }
+    
+    return state
+}
+
+public struct PasskeysRegistrationResponse: Hashable {
+    public var passkeys: [PasskeyRegistration]
+}
+
+extension PasskeysRegistrationResponse: Codable {
+    public enum CodingKeys: String, CodingKey {
+        case passkeys
+    }
+}
+
+
+class PasskeyData {
+    static func fetchPasskeyRegistration() -> Thunk<RowndState> {
```

Comment:
This should be triggered by a listener to updates to Access/Refresh tokens. We can remove all other implementations.

Reply:
i.e., because the ID will be different anyway? would we have any automations that overlap? for example "additional data collection"? i think those won't be device-specific.